### PR TITLE
Update cassandra libuv

### PR DESCRIPTION
**Pull request by richardkchapman**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:
Compiled on OSX and Ubuntu 18.04. Note that this change is required to support ssh 1.1 in Ubuntu 18.04, though it may be possible to use ssh 1.0 instead

Also note that there are now build-time dependencies on libtool, autotools-dev, and automake, so the instructions for building the system will need to be updated. I'll add another commit for that shortly.

Not yet tested usage of cassandra in a built system.

### CODE REVIEW

File: plugins/cassandra/CMakeLists.txt [hunk 1]
```diff
@@ -69,11 +59,12 @@ if(USE_CASSANDRA)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-format-nonliteral") # Work around cassandra build error
     if (CMAKE_COMPILER_IS_GNUCXX)
         if (NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 7.0.0)
-          set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wimplicit-fallthrough=0 -Wno-maybe-uninitialized") # Work around cassandra build error
+          set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wimplicit-fallthrough=0 -Wno-maybe-uninitialized -Wno-deprecated-declarations -Wno-unused-function") # Work around cassandra build error
```
Comment (richardkchapman) [inline plugins/cassandra/CMakeLists.txt]: May not need these flags any more (but they are harmless)

**Comment by richardkchapman** [comment]
@AttilaVamos It's possible that the error is related to new build-time dependencies (see the changes in BUILD_ME.md). Can you investigate?

**Comment by AttilaVamos** [comment]
Yes, I will.

File: plugins/cassandra/CMakeLists.txt [hunk 2]
```diff
@@ -29,27 +29,17 @@ project(cassandraembed)
...
-      COMMAND make builddir_name=${PROJECT_BINARY_DIR}
-      COMMAND mv ${PROJECT_BINARY_DIR}/libuv.so ${PROJECT_BINARY_DIR}/libuv.so.0.10
-      COMMAND ln -s ${PROJECT_BINARY_DIR}/libuv.so.0.10 ${PROJECT_BINARY_DIR}/libuv.so
-      WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}/libuv)
-    add_custom_target(
-      libuv ALL
-      DEPENDS ${PROJECT_BINARY_DIR}/libuv.so)
-    set(LIBUV_LIBRARY ${PROJECT_BINARY_DIR}/libuv.so)
-  endif()
+  add_custom_command(OUTPUT ${PROJECT_BINARY_DIR}/libuv/ COMMAND ${CMAKE_COMMAND} -E make_directory ${PROJECT_BINARY_DIR}/libuv/)
```
[the remark this answers is not in the document]
Comment (richardkchapman) [reply plugins/cassandra/CMakeLists.txt]: It's supposed to be built into {build directory}/plugins/cassandra/libuv/.libs

**Comment by richardkchapman** [comment]
@Michael-Gardner Please review

**Comment by Michael-Gardner** [comment]
@richardkchapman Looks good.  Tested the build on ubuntu 17.10.  I like the use of the CMAKE_SHARED_LIBRARY_SUFFIX variable.  I'll have to start using that elsewhere.

**Comment by richardkchapman** [comment]
@ghalliday Please review/merge

**Pull request merged**